etc/hyperchains/docker-compose-hyperchain-template

```diff
@@ -44,101 +47,88 @@ services:
   zkstack-prover-fri-gateway:
     networks:
       - zkstack
-    build:
-      context: .
-      dockerfile: ./docker/prover-fri-gateway/Dockerfile
+    image: matterlabs/prover-fri-gateway:latest
+    depends_on:
+      zkstack-core:
+        condition: "service_healthy"
     env_file:
-      - ./common.env
-    env:
-      FRI_PROVER_GATEWAY_API_URL: http://zkstack-core:3075
-    ports: # assumes default ports in .env
-      - "3312:3312" # prometheus metrics
+      - {{envFilePath}}
+    environment:
+      FRI_PROVER_GATEWAY_API_URL: http://zkstack-core:3320
+    # ports: # assumes default ports in .env
+    #   - "3312:3312" # prometheus metrics
     volumes:
-      - prover_artifacts:/etc_prover_artifacts
-      - server_artifacts:/etc_server_artifacts
+      - artifacts:{{artifactsPath}}
+      - {{proverSetupArtifacts}}:{{proverSetupArtifacts}}
   zkstack-witness-generator-basic-circuits:
     networks:
       - zkstack
-    build:
-      context: .
-      dockerfile: ./docker/witness-generator/Dockerfile
+    image: matterlabs/witness-generator:latest
```

Comment:
where are these "latest" tags coming from? in this repo we always had "2.0-latest" instead

Reply:
also why are we building some and some are getting pulled from the registry? we need to stay consistent, otherwise we can end up with running multiple versions of code. the same can happen if using only latest from registry, so ideally we want to either pull all witn specific pinned tag (latest semver release deployed and tested to mainnet) or build all from code and dockerfiles

Reply:
Server is built (because is the "easiest" one to change without major protocol changes (and is part of the genesis setup anyways). All prover-related ones are just being pulled for now.

Reply:
decided to fetch and use latest semver in separate/next PR